app/Http/Middleware/AuthenticateAPI.php

```diff
@@ -21,18 +24,53 @@ public function handle($request, Closure $next)
             return response('No auth token detected. Access forbidden.', 403);
         }
         else {
-            /** @var Device $device */
-            $device = Device::with('user')
-                ->where('auth_token', '=', $request->auth_token)
-                ->first();
-
-            if(!$device) {
+            try {
+                $decryptedToken = decrypt($request->auth_token);
+            }
+            catch(Exception $ex) {
                 return response('Invalid auth token. Access forbidden.', 403);
-            } else {
-                $user = $device->user;
-                $user->setActiveDevice($device);
-                Auth::login($user);
-                return $next($request);
+            }
+
+            $explodedToken = explode(TokenConstants::DELIMITER, $decryptedToken);
+
+            $tokenType = $explodedToken[TokenConstants::TYPE_INDEX];
+            $id = $explodedToken[TokenConstants::ID_INDEX];
+            $timestamp = $explodedToken[TokenConstants::TIMESTAMP_INDEX];
+
+            if($tokenType === TokenConstants::DEVICE) {
+                /** @var Device $device */
+                $device = Device::with('user')
+                    ->where([
+                        ['created_at', '=', $timestamp],
+                        ['id', '=', $id]
+                    ])
+                    ->first();
+
+                if(!$device) {
+                    return response('Invalid auth token. Access forbidden.', 403);
+                } else {
+                    $user = $device->user;
+                    $user->setActiveDevice($device);
+                    Auth::login($user);
+                    return $next($request);
+                }
+            }
+            elseif($tokenType === TokenConstants::USER){
+                /** @var User $user */
+                $user = User::
```

Comment:
Same as above, you can use the same repository as above. Just do `$this->repository->setModel(User::class);`